src/NServiceBus.Newtonsoft.Json/NewtonsoftConfigurationExtensions.cs

```diff
@@ -80,6 +116,22 @@ public static void ContentTypeKey(this SerializationExtensions<NewtonsoftSeriali
             config.GetSettings().Set("NServiceBus.Newtonsoft.Json.ContentTypeKey", contentTypeKey);
         }
 
+        /// <summary>
+        /// Configures string to use for <see cref="Headers.ContentType"/> headers.
+        /// </summary>
+        /// <remarks>
+        /// Defaults to <see cref="ContentTypes.Json"/>.
+        /// This setting is required when this serializer needs to co-exist with other json serializers.
+        /// </remarks>
+        /// <param name="config">The <see cref="SerializationExtensions{T}"/> instance.</param>
+        /// <param name="contentTypeKey">The content type key to use.</param>
+        public static void ContentTypeKey(this SerializationExtensions<NewtonsoftJsonSerializer> config, string contentTypeKey)
```

Comment:
Note that all of the overloads that target the original type can be deprecated as well, and they won't need an implementation on master. They can just throw `NotImplementedException`

In some ways, adding the deprecation attribute will just create noise. Imagine the following scenario:

```csharp
var serializer = endpointConfiguration.UseSerializer<NewtonsoftSerializer>();
serializer.ContentTypeKey("Blah");
```

If we deprecate the `ContentTypeKey()` configuration method as well, the user will get two compile time errors, even though they only need to change the first line.

We could maybe resolve this with a marker interface. If we created an interface like `INewtonsoftJsonSerializer` and put it on both serializer definitions then these method signatures could become something like:

```csharp
public static void ContentTypeKey<T>(this SerializationExtensions<T> config, string contentTypeKey)
  where T : INewtonsoftJsonSerializer
```

It does mean adding the marker interface to the public API surface and we'll have to remove that at some point too.

Reply:
I have moved all obsolete methods into their own file and have marked the methods as obsolete to be in line with how we do things in core.